Loading progress bar and real time unfollower view

davidarroyo1234 opened the pull request.

tomleb3 commented:
Nicely done! 😃 
I'll have a better look at this in the following days.
There are some small stuff I think can be improved.

Also, we're running out of room in that top bar 😆 
I've checked and if we merge both this and the other PR with the "copy to clipboard" addition,
it still works but everything gets kinda squshed when the viewport is smaller than 1200px.

This is ok for now but we might want to consider some ways of improving the space situation over there so that future additions would be easier.

tomleb3 commented:
Hey, wondering what did you use to minify the file.

Since if I try to run `npm run build`, which automatically runs prettier and minifies, the HTML template strings come out as multiple lines if I don't put a `\` at the end of each line.

Perhaps we can integrate the tool you're using into the build command instead.

@davidarroyo1234

tomleb3 commented on src/main.js:
```diff
@@ -179,10 +190,11 @@ async function getNonFollowersList(shouldIncludeVerifiedAccounts = true) {
+    const loadingPercentage = getElementByClass('.loading-percentage');
     const elResultsContainer = getElementByClass('.iu_results-container');
     const elNonFollowerCount = getElementByClass('.iu_nonfollower-count');
+    let sleepingContainer = getElementByClass('.sleeping');
+    let sleepingText = getElementByClass('.sleeping-text');
```
No reason to use `let`
IIUC these are not being re-assigned at any point.

tomleb3 commented on src/main.js:
```diff
@@ -239,7 +258,8 @@ window.unfollow = async () => {
         throw new Error('csrftoken cookie is undefined');
     }
-
+    let sleepingContainer = getElementByClass('.sleeping');
+    let sleepingInfo = getElementByClass('.sleeping-text');
```
Also can be consts

tomleb3 commented on src/main.js:
```diff
@@ -147,10 +148,17 @@ function renderOverlay() {
     el.innerHTML = `<header style='position:fixed;top:0;left:0;right:0;display:flex;align-items:center;justify-content:space-between;padding:1rem;height:2.5rem;background-color:#333;z-index:1;'>\
         <div style='font-family:monospace;font-size:1.5em;cursor:pointer;' onclick='location.reload()'>InstagramUnfollowers</div>\
         <label style='display:flex;cursor:pointer;'><input type='checkbox' class='iu_include-verified-checkbox' />&nbsp;Include verified</label>\
+        <div class="progress-bar" style="display:none; width: 120px;height: 30px;border-radius: 5px;margin: 20px 10px;border: 1px solid #7b7777;overflow: hidden;position: relative;">
+            <span class="loading-percentage" style="width: 0;height: 100%;display: block;color: white;line-height: 30px;position: absolute;text-align: end;padding-right: 5px;background-color: #7b7777;">0%</span>
+        </div>
```
Right now it looks a bit weird since the percentage text jumps from left to right as it's climbing.

So here I think we can benefit from having one span for the progress background,
and another one for the percentage.

That way, we can play around with the style to make the percentage span always in the middle.

tomleb3 commented on src/main.js:
```diff
@@ -147,10 +148,17 @@ function renderOverlay() {
         <div style='font-size:1.2em;text-decoration:underline;color:red;cursor:pointer;' onclick='unfollow()'>Unfollow Selected <span class='iu_selected-count'>[0]</span></div>\
         <input type='checkbox' class='iu_toggle-all-checkbox' style='height:1.1rem;width:1.1rem;' onclick='toggleAllUsers(this.checked)' disabled />\
     </header>\
+    <div class="sleeping" style="position:fixed;top:55px;left:0;right:0;display:none;align-items:center;padding:0.7rem;height:1.7rem;background-color:#333;z-index:1;">
+        <hr class="solid" style="border-top: 1px solid #bbb;border-radius: 5px;width: 100%">
+        <label class="sleeping-text" style="color: red;margin-left: 16px;text-align: center"></label>
+    </div>
```
PS -> note that some of the classes were also removed in the example, making it both shorter and more consistent with the other styles, like the topbar.

tomleb3 commented on src/main.js:
```diff
@@ -128,14 +128,15 @@ function renderResults(resultsList) {
             })' />\
         </label>`;
     });
-    getElementByClass('.iu_main-btn').remove();
```
Any reason for doing `display: none;` instead of `.remove()` ?
`.remove()` removes the element from the DOM entirely,
and since there is no use for it later, I think it's slightly better here.

tomleb3 commented on src/main.js:
```diff
@@ -209,22 +221,29 @@ async function getNonFollowersList(shouldIncludeVerifiedAccounts = true) {
             scrollCycle = 0;
-            elResultsContainer.innerHTML =
-                '<span style="margin:1rem;">Sleeping 10 secs to prevent getting temp blocked...</span>';
+            sleepingContainer.style.display = 'inline-grid';
```
Just as a reminder: if we go with the bottom version then this can just be `block` instead.

tomleb3 reviewed (commented):
Just some ideas.
Nice work overall! 💯 💯

davidarroyo1234 commented:
> Hey, wondering what did you use to minify the file.
> 
> Since if I try to run `npm run build`, which automatically runs prettier and minifies, the HTML template strings come out as multiple lines if I don't put a `\` at the end of each line.
> 
> Perhaps we can integrate the tool you're using into the build command instead.
> 
> @davidarroyo1234

I've use a online tool because I though the minifier was broken 😆. I'll check if there is some options or another minifier to achieve this.

davidarroyo1234 commented on src/main.js:
```diff
@@ -128,14 +128,15 @@ function renderResults(resultsList) {
 
 async function run(shouldIncludeVerifiedAccounts) {
+    const elMainBtn = getElementByClass('.iu_main-btn');
+
```
I will take a look to all the comment in the afternoon

davidarroyo1234 commented:
> Hey, wondering what did you use to minify the file.
> 
> Since if I try to run `npm run build`, which automatically runs prettier and minifies, the HTML template strings come out as multiple lines if I don't put a `\` at the end of each line.
> 
> Perhaps we can integrate the tool you're using into the build command instead.
> 
> @davidarroyo1234

I've fixed the comments, and changed the minifier to one thats work with html code. Take a look. After merge you can tweak the css a little bit. Im not to good with that 😆

tomleb3 commented:
> After merge you can tweak the css a little bit

Alright. LGTM!
